L1Trigger/L1THGCal/src/fe_codecs/HGCalBestChoiceCodecImpl.cc

```diff
@@ -80,17 +87,44 @@ HGCalBestChoiceCodecImpl::data_type HGCalBestChoiceCodecImpl::decode(const std::
     return result;
 }
 
+/*****************************************************************/
+void HGCalBestChoiceCodecImpl::linearize(const HGCalTriggerGeometry::Module& mod, const std::vector<HGCEEDataFrame>& dataframes, std::vector<std::pair<HGCEEDetId, uint32_t > >& linearized_dataframes)
+/*****************************************************************/
+{
+    double amplitude; uint32_t amplitude_int;
+    double adcLSB_ =  adcsaturation_/pow(2.,adcnBits_);
+    double tdcLSB_ =  tdcsaturation_/pow(2.,tdcnBits_);
```

Comment:
These don't need to be calculated each time. They can be put in the constructor and adcLSB_ and tdcLSB_ be members of the class.